tests/phpunit/CRM/MembershipExtras/Service/MembershipInstalmentsScheduleTest.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+use CRM_MembershipExtras_Test_Fabricator_MembershipType as MembershipTypeFabricator;
+use CRM_MembershipExtras_Service_MembershipInstalmentsSchedule as MembershipInstalmentsSchedule;
+use CRM_MembershipExtras_Exception_InvalidMembershipTypeInstalmentCalculator as InvalidMembershipTypeInstalmentCalculator;
+use CRM_MembershipExtras_Service_MoneyUtilities as MoneyUtilities;
+
+/**
+ * Class CRM_MembershipExtras_Service_MembershipTypeInstalmentsScheduleTest
+ *
+ * @group headless
+ */
+class CRM_MembershipExtras_Service_MembershipTypeInstalmentsScheduleTest extends BaseHeadlessTest {
+
+  /**
+   * Tests Generate Rolling Monthly Instalments
+   *
+   * @throws Exception
+   */
+  public function testGenerateRollingMonthlyInstalments() {
+    $membershipTypes = $this->getRollingMembershipTypes();
+    $instalments = $this->getRollingMembershipInstalments(
+      $membershipTypes,
+      MembershipInstalmentsSchedule::MONTHLY
+    );
+
+    $this->assertCount(MembershipInstalmentsSchedule::MONTHLY_INSTALMENT_COUNT, $instalments);
+
+    $expectedAmount = MoneyUtilities::roundToPrecision(
+      ($membershipTypes[0]->minimum_fee + $membershipTypes[1]->minimum_fee) / MembershipInstalmentsSchedule::MONTHLY_INSTALMENT_COUNT,
+      2
+    );
+
+    $instalment = $instalments[0];
+    $this->assertEquals($expectedAmount, $instalment->getInstalmentAmount()->getAmount());
+  }
+
+  /**
+   * Test Generate Rolling Quarterly Instalment
+   *
+   * @throws Exception
+   */
+  public function testGenerateRollingQuarterlyInstalments() {
+    $membershipTypes = $this->getRollingMembershipTypes();
+    $instalments = $this->getRollingMembershipInstalments(
+      $membershipTypes,
+      MembershipInstalmentsSchedule::QUARTERLY
+    );
+
+    $this->assertCount(MembershipInstalmentsSchedule::QUARTERLY_INSTALMENT_COUNT, $instalments);
+
+    $expectedAmount = MoneyUtilities::roundToPrecision(
+      ($membershipTypes[0]->minimum_fee + $membershipTypes[1]->minimum_fee) / MembershipInstalmentsSchedule::QUARTERLY_INSTALMENT_COUNT,
+      2
+    );
+
+    $instalment = $instalments[0];
+    $this->assertEquals($expectedAmount, $instalment->getInstalmentAmount()->getAmount());
+  }
+
+  /**
+   * Test Generate Rolling Annually Instalment
+   *
+   * @throws Exception
+   */
+  public function testCalculateRollingAnnuallyInstalments() {
```

Comment:
you used `generate` in the above test methods and this should follow that, but anyway in my comment above I suggested to just remove it.